### ISSUE TITLE
TTY support

### ISSUE THREAD

**Issue by kardwen** [kardwen/passepartui#9]
I created this issue to bundle the efforts to properly support the use of passepartui in a virtual console.

Here is a preliminary list of tasks that could be done:

* Solution for pinentry in TTY
* Alternative widget mode without Unicode characters
* TTY color theme

**Comment by kardwen** [comment]
I also noticed that the screen is not cleared automatically on exit in VC.

**Comment by kardwen** [comment]
I investigated the issue and found that the problem is that `passepartui` and in my case `pinentry-curses` are attempting to write to the terminal at the same time as the `pass` operation is running in a new thread. I have experimented a bit with `gpgme` to first request a passphrase and then perform the `pass` operation asynchronously, which works but seems rather fragile.

In conclusion, I would prefer to "rewrite everything in Rust" using `sequoia-openpgp` instead of introducing `gpgme` as a new dependency. This would also have some other advantages...

**Comment by kardwen** [comment]
I have re-evaluated the changes and decided to use `gpgme` for `passepartui` because the implementation is relatively straightforward and `sequoia-openpgp` would need a compatibility layer to GnuPG. I dislike that this requires a system library but `gpgme` is probably already installed on most systems.

I still have some other ideas for supporting sequoia-pgp in the future...

**Comment by kardwen** [comment]
Yes. I realized what you meant when I tried it myself with `pinentry-curses`, which lead to really weird behaviour. I hoped that I could find an option to disable only terminal pinentry programs in `gpgme` but there isn't one.

So I decided to add an option that switches to blocking calls to `gpgme` for decryption operations, just run:

```sh
passepartui --tty-pinentry
```

This also triggers a rerender of the entire TUI after each decryption operation. I know that this is not a perfect solution, it should be an option in a configuration file.

@grouzen I would be happy if you could evaluate it, maybe you can come up with something that I have overlooked. I already published a new release because I did quite a lot of changes in https://github.com/kardwen/passepartout

**Comment by grouzen** [comment]
Hey! Just found some time to try it out. Works like a charm! Thank you!

**Comment by kardwen** [comment]
Nice, thank you for the report!
